app/scripts/directives/miniLog.js

```diff
@@ -0,0 +1,105 @@
+'use strict';
+
+angular.module('openshiftConsole').component('miniLog', {
+  controllerAs: 'miniLog',
+  controller: [
+    '$scope',
+    '$filter',
+    'APIService',
+    'DataService',
+    'HTMLService',
+    MiniLogController
+  ],
+  bindings: {
+    apiObject: '<',
+    numLines: '<',
+    context: '<'
+  },
+  templateUrl: 'views/overview/_mini-log.html'
+});
+
+function MiniLogController($scope,
+                           $filter,
+                           APIService,
+                           DataService,
+                           HTMLService) {
+  var miniLog = this;
+
+  var name, logSubresource, streamer;
+  var annotation = $filter('annotation');
+  var numLines = miniLog.numLines || 7;
+  var buffer = [];
+  miniLog.lines = [];
+
+  var update = _.throttle(function() {
+    $scope.$evalAsync(function() {
+      miniLog.lines = _.clone(buffer);
+    });
+  }, 200);
+
+  // Used as a "track by" ID in the view. Since we follow from the end, might
+  // not correspond to actual line numbers. Track by is necessary in case a
+  // line is repeated.
+  var lineNum = 0;
+  var onMessage = function(msg) {
+    if (!msg) {
+      return;
+    }
+
+    var escaped = ansi_up.escape_for_html(msg);
+    var html = ansi_up.ansi_to_html(escaped);
+    var linkifiedHTML = HTMLService.linkify(html, '_blank', true);
+
+    lineNum++;
+    buffer.push({
+      markup: linkifiedHTML,
+      id: lineNum
+    });
+
+    if (buffer.length > numLines) {
+      buffer = _.takeRight(buffer, numLines);
```

Comment:
Yeah, not sure what's best. At least in this case, the buffer is max 8 lines, so we aren't ever keeping too much data in memory. And we're only calling `_.takeRight` on a small array.